spec/components/chart_spec.rb

```diff
@@ -0,0 +1,96 @@
+require 'rails_helper'
+
+data = {
+  caption: "Unique page views",
+  h_axis_title: "Date",
+  v_axis_title: "Views",
+  chart_title: "Page views chart",
+  table_title: "Page views table",
+  chart_id: 'pageviews_jan_chart',
+  table_id: 'pageviews_jan_table',
+  keys: %w(Jan Feb Mar Apr May Jun Jul Aug Sep Oct Nov Dec),
+  rows: [
+    {
+      label: "2017",
+      values: [5, 119, 74, 117, 50, 119, 61, 110, 12, 21, 121, 67]
+    },
+    {
+      label: "2018",
+      values: [3, 8, 37, 435, 78, 4, 9, 61, 110, 12, 21, 121]
+    }
+  ]
+}
+describe "Chart", type: :view do
+  def render_component(locals)
+    render partial: "components/#{component_name}", locals: locals
+  end
+
+  def component_name
```

Comment:
[It feels better to use `let`](https://relishapp.com/rspec/rspec-core/v/3-8/docs/helper-methods/let-and-let)